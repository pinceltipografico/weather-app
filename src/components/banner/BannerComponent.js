import React from 'react';
import PropTypes from 'prop-types';
import styles from './BannerComponent.scss';
import { imagesMap } from '../../constants/imagemap';
import dayjs from 'dayjs';

export const BannerComponent = (props) => {
  const {todayWeather} = props;
  const date = dayjs(todayWeather.dt_text).format('MMMM DD, HH:hh');
  const weather = todayWeather.weather[0];
  const {temp_max, temp_min, temp, humidity} = todayWeather.main;
  const wallPaper = imagesMap[weather.id].wallPaper || props.defaultWallPaper;
  const icon = imagesMap[weather.id].icon;
  
  return (
    <section className={styles.banner}
      style={{
        backgroundImage: `url(${wallPaper})`
      }}>
      <div className={styles['weather-data']}>
        <span className={styles.date}>{props.city} - {date}</span>
        <div className={styles.widget}>
          <div>
            <span>Day {temp_max}&#176; - Night {temp_min}&#176;</span>
            <h1>{temp}<small>&#176;F</small></h1>
            <span>air humidity {humidity}</span>
          </div>

          {/* it would be cool draw some custom icons here */}
          {/* because the provider icon from the api is so small */}
          <div className={styles.icon}>
            <img src={icon} alt="mostly sunny"/>
            <span>{weather.description}</span>
          </div>

        </div>
      </div>
    </section>
  )
}

BannerComponent.defaultProps = {
  defaultWallPaper: imagesMap[801].wallPaper,
  todayWeather: null
};

BannerComponent.propTypes = {
  city: PropTypes.string.isRequired,
  defaultWallPaper: PropTypes.string.isRequired,
  todayWeather: PropTypes.object
};