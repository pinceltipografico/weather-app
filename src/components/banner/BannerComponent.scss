@import '~@/scss/mixins';

$white: #fff;

.banner {
  align-items: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  height: rem(500px);
  margin-top: rem(100px);
  overflow: hidden;
  position: relative;
  width: 100%;
}

.weather-data {
  display: flex;
  flex-direction: column;
  padding: rem(40px);
  width: 100%;
}

.date {
  @include font-size(1.5);
  color: $white;
  margin: 0 auto;
  max-width: 1024px;
  width: 100%;
}

.widget {
  align-items: center;
  display: flex;
  margin: 0 auto;
  max-width: 1024px;
  padding-top: rem(40px);
  width: 100%;

  span {
    @include font-size(1.2);
    color: $white;
  }

  h1 {
    @include font-size(10);
    color: $white;
    margin: 0;
    position: relative;

    // brea for small phones
    @include breakpoint(phone-wide, max) {
      @include font-size(8);
    }
  }

  small {
    @include font-size(5);
    font-weight: 100;
    position: absolute;
    right: rem(-70px);
    top: rem(15px);
  }
}

.icon {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-left: auto;

  img {
    max-width: rem(300px);
  }

  @include breakpoint(phone-wide, max) {
    display: none;
  }
}
