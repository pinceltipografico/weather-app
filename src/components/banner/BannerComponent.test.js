import React from 'react';
import { shallow, mount } from 'enzyme';
import { BannerComponent } from './BannerComponent';

describe('BannerComponent', () => {
  let banner = null
  let props = {}

  beforeEach(() => {
    props.city = 'SÃ£o Paulo';
    props.wallPaper = 'cloudy.jpg';
    props.todayWeather = {
      "dt": 1552575600,
      "main": {
        "temp": 32.74,
        "temp_min": 28.48,
        "temp_max": 32.74,
        "pressure": 1029.69,
        "sea_level": 1029.69,
        "grnd_level": 993.33,
        "humidity": 84,
        "temp_kf": 2.36
      },
      "weather": [
        {
          "id": 803,
          "main": "Clouds",
          "description": "broken clouds",
          "icon": "04d"
        }
      ],
      "clouds": {
        "all": 68
      },
      "wind": {
        "speed": 2.15,
        "deg": 70.0042
      },
      "rain": {},
      "sys": {
        "pod": "d"
      },
      "dt_txt": "2019-03-14 15:00:00"
    };
  })

  const mountedBanner = () => {
    if(!banner) {
      banner = mount(<BannerComponent {...props}/>)
    }
    return banner
  }

  it('should render without problems', () => {
    const component = shallow(<BannerComponent {...props}/>)
    expect(component).toMatchSnapshot()
  })

  it('should render default image on banner', () => {
    const section = mountedBanner().find('section').first()
    expect(section.props().style.backgroundImage).toBe(`url(${props.wallPaper})`)
  })

})