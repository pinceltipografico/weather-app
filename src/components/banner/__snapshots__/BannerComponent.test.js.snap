// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerComponent should render without problems 1`] = `
<section
  style={
    Object {
      "backgroundImage": "url(cloudy.jpg)",
    }
  }
>
  <div>
    <span>
      S찾o Paulo
       - 
      March 14, 10:10
    </span>
    <div>
      <div>
        <span>
          Day 
          32.74
          째 - Night 
          28.48
          째
        </span>
        <h1>
          32.74
          <small>
            째F
          </small>
        </h1>
        <span>
          air humidity 
          84
        </span>
      </div>
      <div>
        <img
          alt="mostly sunny"
          src="http://openweathermap.org/img/w/04d.png"
        />
        <span>
          broken clouds
        </span>
      </div>
    </div>
  </div>
</section>
`;
